vault/resource_secrets_sync_association.go

```diff
@@ -0,0 +1,182 @@
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+const (
+	fieldSecretName = "secret_name"
+	fieldSyncStatus = "sync_status"
+	fieldAccessor   = "accessor"
+	fieldUpdatedAt  = "updated_at"
+)
+
+var secretsSyncAssociationFields = []string{
+	fieldSecretName,
+	fieldSyncStatus,
+	fieldAccessor,
+	fieldUpdatedAt,
+}
+
+func secretsSyncAssociationResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: provider.MountCreateContextWrapper(secretsSyncAssociationWrite, provider.VaultVersion115),
+		ReadContext:   provider.ReadContextWrapper(secretsSyncAssociationRead),
+		DeleteContext: secretsSyncAssociationDelete,
+
+		Schema: map[string]*schema.Schema{
+			consts.FieldName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the store/destination.",
+				ForceNew:    true,
+			},
+			consts.FieldType: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Type of secrets store.",
+				ForceNew:    true,
+			},
+			consts.FieldMount: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Specifies the mount where the secret is located.",
+			},
+			fieldAccessor: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "Specifies the unique accessor of the mount.",
+				ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+			},
+			fieldSecretName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Specifies the name of the secret to synchronize.",
+			},
+			fieldSyncStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Specifies the status of the association.",
+			},
+			fieldUpdatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Duration string stating when the secret was last updated.",
+			},
+		},
+	}
+}
+
+func secretsSyncAssociationWrite(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	name := d.Get(consts.FieldName).(string)
+	destType := d.Get(consts.FieldType).(string)
+	path := secretsSyncAssociationSetPath(name, destType)
+
+	data := map[string]interface{}{}
+
+	for _, k := range []string{
+		fieldSecretName,
+		consts.FieldMount,
+	} {
+		data[k] = d.Get(k)
+	}
+
+	log.Printf("[DEBUG] Writing association to %q", path)
+	resp, err := client.Logical().WriteWithContext(ctx, path, data)
+	if err != nil {
+		return diag.Errorf("error setting secrets sync association %q: %s", path, err)
+	}
+	log.Printf("[DEBUG] Wrote association to %q", path)
+
+	// expect accessor to be provided from mount
+	accessor := d.Get(fieldAccessor).(string)
+
+	// associations can't be read from Vault
```

Comment:
Just for clarification associations can be read. E.g.:
```
http://127.0.0.1:8200/v1/sys/sync/associations/destinations?mount=:mount&secret_name=:name
```

Though I agree that using the data retuned from the write request like you are doing is the best approach; an explicit read request is not necessary.